manticore/backend/compiler.py

```diff
@@ -70,11 +70,18 @@ def __as_replace(self, pk):
         for field, _, value in self.query.values:
             fields.append(field)
             setattr(obj, field.attname, value)
-        if set(fields) ^ set(opts.local_fields):
-            # If you call REPLACE omitting some attributes, manticore will
-            # store default empty values (0, '', [] etc...)
```

Comment:
Have this manticore behavior changed? 

```sql
insert into tbl(id, a, b) values(1, "a","b");
replace into tbl (id, a) values(1, "A");
select * from tbl where id=1;
-- (1, "a", "")
```

This does not look like Tbl.objects.filter(pk=1).update(a='A')

Reply:
That's why on line 79 of the code, we retrieve all the missing field values from the existing object and add them to the replace query.

```
           current_obj = self.query.model.objects.get(pk=pk)
            for field in excluded_fields:
                value = getattr(current_obj, field.attname)
                fields.append(field)
                setattr(obj, field.attname, value)
```

This is also applicable to previous versions, but it causes an additional request. Any ideas on how to make it more optimal?

Reply:
Just let user call save() without update_fields instead, it is too dangerous to update python object from db in general.

Reply:
Here I revert. 

And here I added.
https://github.com/just-work/django-manticore/pull/28/files#diff-f627e6d41d572ae0be0c5fa99b7703f0b893f2a1ca1a3cf695fcbf5d03e6ad66R83

Reply:
Can we just leave it unsupported?

I don't want to choose between changing django-manticore behavior and django (itself) behavior.

I mean, if Django now adds update_fields to update_or_create, there is a reason, documentation and some user expectations on that.
On the manticore (database) side, the problem definition is:

* UPDATE can update only fixed-length fields, because it is table row in-place change
* REPLACE can't do partial update, because it is whole table row del+add

With manticore, user just can't do partial updates in some cases (text, binary, multifield updates), and can in other cases.

My proposal is to leave old behavior as-is (at least, it does not make worth than unsupported django-4.2), and add a separate issue to more thoroughly split partial updates to supported by database and unsupported by database.

Reply:
@tumb1er I fixed the test. Now it works.